### ISSUE TITLE
[a11y]: ai-chat should allow arrow key navigation to messages with JAWS

### ISSUE THREAD

**Issue by Daniel-Schulz-Private** [carbon-design-system/carbon-labs#876]
### Package version

0.18.0

### React version

18.3.1

### Automated testing tool and ruleset

n/a

### Assistive technology

JAWS

### Description

In the AI Chat window, neither the user’s submitted queries nor the AI’s responses are readable using standard arrow key navigation. Screen reader users are unable to move through or read the conversation content , on Arrow key navigation.
- User queries and AI responses are not read out with arrow keys.
- Interactive elements within the chat, such as links and buttons, are also not or announced correctly.
As a result, chat window is completely in accessible for Screen reader users on Arrow key navigation.
Note : Screen reader identified a Tree view structure in Chat Window , which adding additional complication, on user experience .

Recommended fix: Make sure that the chat content can be navigated and red line by line or element by element using screen reader arrow key navigation:

Ensure that all chat messages (both user queries and AI responses) are presented in a way that is readable by screen readers and conveyed in the same sequential order as visually displayed.
Make interactive elements within chat responses (such as links and buttons) fully keyboard-accessible, and ensure they expose correct roles, names, and states to assistive technologies.
User impact: If the chat window is not readable or navigable, the entire chat content becomes inaccessible for screen reader users. This prevents them from reviewing user queries, AI responses, and interactive elements, resulting in a complete loss of access to the conversation.

Related to #380.

### WCAG 2.1 Violation

WCAG Mapping: 1.3.1 Info and Relationships

### Reproduction/example

n/a

### Steps to reproduce

1. make sure JAWS is enabled
2. open ai-chat
3. ask question
4. wait for answer
5. try to key-navigate to question and answer
6. see that both are inaccessible using JAWS

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon-labs/blob/main/.github/CODE_OF_CONDUCT.md)
- [x] I checked the [current issues](https://github.com/carbon-design-system/carbon-labs/issues) for duplicate problems
